soteria-rust/lib/builtins/miri.ml

```diff
@@ -7,7 +7,7 @@ module M (Heap : Heap_intf.S) = struct
 
   type nonrec rust_val = Sptr.t rust_val
 
-  let alloc_id ~args ~state =
```

Comment:
Would it be worth defining InterpM outside Interp.ml so you can re-use it in miri/core/std?